src/Microsoft.Tye.Hosting/Model/Application.cs

```diff
@@ -25,6 +27,8 @@ public Application(FileInfo source, Dictionary<string, Service> services)
 
         public Dictionary<string, Service> Services { get; }
 
+        internal ConcurrentDictionary<string, TaskCompletionSource<ProcessResult>> ProjectProcesses { get; } = new ConcurrentDictionary<string, TaskCompletionSource<ProcessResult>>();
```

Comment:
What is this?

Reply:
In case there exists multiple services that are based on the same project file, we want to trigger the build only once, instead of building the same project multiple times. I guess I can probably name this better so it's more obvious what this is.